.github/workflows/hadolint-ci.yml

```
@@ -0,0 +1,20 @@
+name: 'Hadolint Linting'
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  hadolint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install hadolint
+        run: sudo wget -O /bin/hadolint https://github.com/hadolint/hadolint/releases/download/v2.7.0/hadolint-Linux-x86_64 && sudo chmod +x /bin/hadolint
+
+      - name: Lint Dockerfiles
+        run: hadolint **/Dockerfile**
```

I don't think there's a need for such manual installation. We can use the hadolint container.
```suggestion
  lint:
    runs-on: ubuntu-latest
    container: hadolint/hadolint:latest-debian
    steps:
    - uses: actions/checkout@v2
    - name: hadolint
      run: hadolint **/*Dockerfile*
```

the hadolint action should also work https://github.com/hadolint/hadolint-action

see the draft PR https://github.com/ChainSafe/forest/pull/2615/files (you may need to enable recursive search)